D-OS Save Editor/savegame/Meta.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace D_OS_Save_Editor
+{
+    public class Meta
+    {
+        public enum DifficultyEnum
+        {
+            Story = 0,
+            Classic,
+            Tactician,
+            Honour
+        }
+        public enum SaveGameTypeEnum
+        {
+            Manual = 0,
+            QuickSave,
+            AutoSave,
+            HonourSave
+        }
+        private string _level;
+        private int _difficulty;
+        private List<string> _gameVersion;
+        private int _saveGameType;
+        private string _seed;
+        private string _saveTime;
+        private List<string> _modsName;
+
+        //<node id="MetaData">
+
+        //<attribute id = "Level" value="Cyseal" type="22" />
+        /// <summary>
+        /// Game map/level
+        /// </summary>
+        public string Level { get => _level; set => _level = value; }
+        /// <summary>
+        /// Difficulty Game
+        /// </summary>
+        public int Difficulty { get => _difficulty; set => _difficulty = value; }
+        /// <summary>
+        /// Full game version
+        /// </summary>
+        public List<string> GameVersion { get => _gameVersion; set => _gameVersion = value; }
+        /// <summary>
+        /// Date of savegame created
+        /// </summary>
+        public string SaveTime { get => _saveTime; set => _saveTime = value; }
+        /// <summary>
+        /// List of Mod's Name load in game
+        /// </summary>
+        public List<string> ModsName { get => _modsName; set => _modsName = value; }
+        //<attribute id="SaveGameType" value="2" type="27" />
+        /// <summary>
+        /// The type of SaveGame
+        /// </summary>
+        public int SaveGameType { get => _saveGameType; set => _saveGameType = value; }
+        public string Seed { get => _seed; set => _seed = value; }
+
+        public Meta()
+        {
+            GameVersion = new List<string>();
+            ModsName = new List<string>();
+        }
+
+        public void SaveTimeRead(int year, string month, string day, string hours, string minutes, string secondes, int milliseconds)
+        {
+            SaveTime = Convert.ToString(1900 + year) + '/' + month.PadLeft(2, '0') + '/' + day.PadLeft(2, '0') + ' ' + hours.PadLeft(2, '0') + ':' + minutes.PadLeft(2, '0') + ':' + secondes.PadLeft(2, '0');
```

Comment:
If the `year`, `month` etc are defined separately in XML, I would probably define and use a `SaveTime` class. Inside the class, you have properties `year`, `month` etc, and then override the `ToString` method ([How to: Override the ToString Method (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/how-to-override-the-tostring-method)).

Reply:
Good idea. I wrote this to have a string return with less work (to save time which lost now 🤣 ) I work on that and I'll come back ^^

Reply:
The only problem is it doesn't read well, because you are read so many input arguments and then the method returns nothing. Instead of using a separate class, you could also implement these `year` and `month` properties in the `Meta` class and the `get` method for `SaveTime` can implement this conversion. This is probably better.